### PR TITLE
Use guard statement for early exits in removeAll() in HashedCollections.swift.gyb

### PR DESCRIPTION
**Pull request by florianreinhart**

### CODE REVIEW

File: stdlib/public/core/HashedCollections.swift.gyb [hunk 1]
```diff
@@ -3413,16 +3413,16 @@ internal enum _Variant${Self}Storage<${TypeParametersDecl}> : _HashStorageType {
...
     }
 
-    if !keepCapacity {
+    guard keepCapacity else {
       self = .Native(NativeStorage.Owner(minimumCapacity: 2))
       return
     }
 
-    if _fastPath(guaranteedNative) {
+    guard !_fastPath(guaranteedNative) else {
```
Comment (westerlund) [inline stdlib/public/core/HashedCollections.swift.gyb]: This doesn't make any sense. It's actually harder to read because of the double negative.

**Comment by gribozavr** [comment]
I agree with @westerlund, the proposed version is harder to read.

**Comment by florianreinhart** [comment]
I agree that in this particular line it is harder to read. However, the nice thing about `guard` is that the compiler enforces a return statement.

Is there a guideline when you should use `guard` or `if` for early exits? I assumed it is bad style to use `if` for early exits.

**Comment by gribozavr** [comment]
Definitely use `guard` for early exists when you are unwrapping an optional -- that allows you to bind the unwrapped value to a local variable.  In other cases it is frequently the case that one ends up with a double negative.

**Pull request closed without merging**